### ISSUE TITLE
e2e flake: TestS3DisasterRecovery

### ISSUE THREAD

**Issue by hongchaodeng** [coreos/etcd-operator#534]
```
etcd-operator logs END ===
--- FAIL: TestS3DisasterRecovery (50.42s)
    --- FAIL: TestS3DisasterRecovery/disaster_recovery_on_S3 (0.00s)
        --- FAIL: TestS3DisasterRecovery/disaster_recovery_on_S3/2_members_(majority)_down (50.42s)
        	e2e_test.go:242: failed to create 3 members etcd cluster: still failing after 6 retries
        --- FAIL: TestS3DisasterRecovery/disaster_recovery_on_S3/3_members_(all)_down (50.42s)
        	e2e_test.go:242: failed to create 3 members etcd cluster: still failing after 6 retries
```

**Comment by hongchaodeng** [comment]
In fact, the failure has nothing to do with disaster recovery. It couldn't even start the initial cluster.

**Comment by hongchaodeng** [comment]
From the logs, right after controller has started, it just thinks "no etcd members exist":
```
start watching at 104061
all etcd pods are dead. Trying to recover from a previous backup
```

A possible path is that it goes into cluster.Restore(), which doesn't create seed member.

There could be a race that:
- etcd cluster TPR created
- controller starts watching

**Comment by hongchaodeng** [comment]
Closed via #541
